plugins/src/main/java/org/smartregister/fhir/gateway/plugins/PermissionAccessChecker.java

```diff
@@ -102,7 +103,15 @@ public AccessDecision checkAccess(RequestDetailsReader requestDetails) {
 
     private void initSyncAccessDecision(RequestDetailsReader requestDetailsReader) {
         Map<String, List<String>> syncStrategyIds;
-
+        String clientRole = Utils.getClientRole(userRoles, logger);
+
+        if (!clientRole.equals(Constants.ROLE_ANDROID_CLIENT)) {
+            syncStrategyIds = Collections.emptyMap();
+            this.syncAccessDecision =
+                    new SyncAccessDecision(
+                            fhirContext, jwt.getSubject(), true, syncStrategyIds, "", userRoles);
+            return;
+        }
         Composition composition = fetchComposition();
         String syncStrategy = readSyncStrategyFromComposition(composition);
```

Comment:
@Rkareko This still seems to fetch composition.  This should only happen for users with the `ANDROID_CLIENT` role